### PR TITLE
add pinecone db indexer

### PR DESCRIPTION
**Pull request by shashank-reddy-code**
fetches and parses a given github repo to index in a pinecone vector db from openai embeddings

### CODE REVIEW

File: prompts-service/pinecone_db.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+pinecone.init(
+    api_key=os.environ['PINECONE_API_KEY'],
+    environment='asia-southeast1-gcp-free'
+)
+
+def index_repo(repo: str):
+    namespace = extract_repo_name(repo)
+    vector_store = Pinecone(
+        index=pinecone.Index('pinecone-index'),
```
Comment (anirudhsaligram) [inline prompts-service/pinecone_db.py:31]: Lets make this `pinecode-index` as a constant as its used in other parts of the file.

File: prompts-service/pinecone_db.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    
+    # todo: convert repo url to zip file url
+    with zipfile_from_github(zip_repo_url=repo) as zip_ref:
+        zip_file_list = zip_ref.namelist()
+        
+        pbar = tqdm(zip_file_list, desc=f'Total tokens: 0')
+        file_prefix = extract_file_prefix(repo)
+        for file_name in pbar:
+            # todo: exclude more files based on language of the repo and make this customizable in the request
+            if (file_name.endswith('/') or
```
Comment (anirudhsaligram) [inline prompts-service/pinecone_db.py:51]: Refactor this to move the `file_name` checks into a helper method and call that here.

File: prompts-service/pinecone_db.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    # todo: use better splitter that is meant for code
+    split_documents = splitter.create_documents(file_texts, metadatas=metadatas)
+    vector_store.from_documents(
+        documents=split_documents, 
+        embedding=EMBEDDINGS,
+        index_name='pinecone-index',
+        namespace=namespace
+    )
+
+    pd.DataFrame.from_records(corpus_summary).to_csv('data/corpus_summary.csv', index=False)
```
Comment (anirudhsaligram) [inline prompts-service/pinecone_db.py:78]: Add a debug flag and export the csv only when debug flag is set to `true`

**Review by anirudhsaligram — COMMENTED** [review]
LGTM  in general 👍🏼 
Minor nits only

**Review by anirudhsaligram — COMMENTED** [review]
LGTM  in general 👍🏼 
Minor nits only

**Comment by shashank-reddy-code** [comment]
@review-buddy

**Pull request closed without merging**